DotNet/Shared/Application/Error/Handlers/DeadLetterExceptionHandler.cs

```diff
@@ -32,33 +31,34 @@ public void HandleException(ConsumeResult<K, V> consumeResult, string facilityId
         {
             try
             {
-                Logger.LogError(message: $"{GetType().Name}: Failed to process {ServiceName} Event.", exception: new Exception(message));
+                var ex = new Exception(message);
+                Logger.LogError(ex, "{Name}: Failed to process {S} Event.", GetType().Name, ServiceName);
 
                 ProduceDeadLetter(consumeResult.Message.Key, consumeResult.Message.Value, consumeResult.Message.Headers, message);
             }
             catch (Exception e)
             {
-                Logger.LogError(e, $"Error in {GetType().Name}.HandleException: " + e.Message);
+                Logger.LogError(e, "Error in {name}.HandleException: {message}", GetType().Name, e.Message);
             }
         }
 
         public virtual void HandleException(ConsumeResult<K, V> consumeResult, Exception ex, string facilityId)
         {
-            var dlEx = new DeadLetterException(ex.Message, ex.InnerException);
+            var dlEx = new DeadLetterException(ex.Message, ex.InnerException ?? ex.GetBaseException());
```

Comment:
Maybe not as problematic as the above, but don't we want *this* exception (i.e., `ex` itself) to be the `DeadLetterException`'s inner exception?